providers/implementations/kdfs/tls1_prf.c

```diff
@@ -194,13 +207,40 @@ static int fips_ems_check_passed(TLS1_PRF *ctx)
     if (!ems_approved) {
         if (!OSSL_FIPS_IND_ON_UNAPPROVED(ctx, OSSL_FIPS_IND_SETTABLE0,
                                          libctx, "TLS_PRF", "EMS",
-                                         ossl_tls1_prf_ems_check_enabled)) {
+                                         FIPS_tls_prf_ems_check)) {
             ERR_raise(ERR_LIB_PROV, PROV_R_EMS_NOT_ENABLED);
             return 0;
         }
     }
     return 1;
 }
+
+static int fips_digest_check_passed(TLS1_PRF *ctx)
```

Comment:
There should be no reason why you cant do this check when it sets the digest in the set_ctx_params()..

Reply:
See https://github.com/openssl/openssl/discussions/24904#discussioncomment-10067515

Reply:
Do you mean that checking digest in set_ctx_param and making indicator be unapproved directly ?

Reply:
I have updated the implementation. We check the digest in set_ctx_params now.